core/drand_interceptors.go

```diff
@@ -0,0 +1,80 @@
+package core
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/drand/drand/protobuf/common"
+	"github.com/drand/drand/utils"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type ContextGetter interface {
+	GetContext() *common.Context
+}
+
+func (d *Drand) NodeVersionValidator(ctx context.Context, req interface{},
```

Comment:
I understand now you have chosen to put the context in every messages and automatically validate if it comes from a compatible version or not.

Reply:
For me this is a bit problematic:
1. It makes again a huge PR very long to review
2. It assumes everything is not compatible from different major version while that's not necessarily true (like public key can stay the same format but the randomness beacon is not etc)
3. It's very very redundant, like usually we check before doing any interaction with another node its version and then we go on. (think TLS negotiation) We could simply have an endpoint Version()  that is checked automatically before dkg/resharing to see if everyone has compatible versions. Now every messages has this, it's much heavier on the network (even if not by a lot I agree) and most importantly on the codebase itself.